src/App.tsx

```diff
@@ -564,223 +692,250 @@ function App() {
-                              newAddress,
-                              scope as `${string}:${string}`,
+                          : scope}
+                      </h3>
+
+                      <select
+                        className="accounts-select"
+                        value={selectedAccounts[scope] ?? ''}
+                        onChange={(evt) => {
+                          const newAddress =
+                            (evt.target
+                              .value as `${string}:${string}:${string}`) ?? '';
+                          setSelectedAccounts((prev) => ({
+                            ...prev,
+                            [scope]: newAddress,
+                          }));
+
+                          // modify the method request if it's a signing method (inject selected address)
+                          const currentMethod = selectedMethods[scope];
+                          if (
+                            currentMethod &&
+                            currentMethod in SIGNING_METHODS
+                          ) {
+                            const example =
+                              metamaskOpenrpcDocument?.methods.find(
+                                (method) =>
+                                  (method as MethodObject).name ===
+                                  currentMethod,
+                              );
+
+                            if (example) {
+                              let exampleParams: Json = openRPCExampleToJSON(
+                                example as MethodObject,
+                              );
+
+                              // inject the newly selected address
+                              exampleParams = insertSigningAddress(
+                                currentMethod,
+                                exampleParams,
+                                newAddress,
+                                scope as `${string}:${string}`,
+                              );
+
+                              const updatedRequest = {
+                                method: 'wallet_invokeMethod',
+                                params: {
+                                  scope,
+                                  request: exampleParams,
+                                },
+                              };
+
+                              setInvokeMethodRequests((prev) => ({
+                                ...prev,
+                                [scope]: JSON.stringify(
+                                  updatedRequest,
+                                  null,
+                                  2,
+                                ),
+                              }));
+                            }
+                          }
+                        }}
+                      >
+                        <option value="">Select an account</option>
+                        {(details.accounts ?? []).map(
+                          (account: `${string}:${string}:${string}`) => {
```

Comment:
worth using the utils package type here?

Reply:
yep! done here: https://github.com/MetaMask/test-dapp-multichain/pull/18/commits/b94a978674bf0b86a8851683e7e30d248bc4ef96